great_expectations/rule_based_profiler/util.py

```diff
@@ -46,34 +39,72 @@ def get_validator(
             f"{expectation_suite_name}_{domain.id}_suite_{str(uuid.uuid4())[:8]}"
         )
 
-    return data_context.get_validator(
-        batch_request=batch_request,
-        create_expectation_suite_with_name=expectation_suite_name,
-    )
+    batch: Batch
+    if batch_list is None or all([batch is None for batch in batch_list]):
+        if batch_request is None:
+            return None
+
+        batch_request = build_batch_request(
+            domain=domain,
+            batch_request=batch_request,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        validator = data_context.get_validator(
+            batch_request=batch_request,
+            create_expectation_suite_with_name=expectation_suite_name,
+        )
+    else:
+        num_batches: int = len(batch_list)
+        if num_batches == 0:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"""{__name__}.get_validator() must utilize at least one Batch ({num_batches} are available).
+"""
+            )
+
+        expectation_suite: ExpectationSuite = data_context.create_expectation_suite(
+            expectation_suite_name=expectation_suite_name
+        )
+        validator = data_context.get_validator_using_batch_list(
+            expectation_suite=expectation_suite,
+            batch_list=batch_list,
+        )
+
+    return validator
 
 
 def get_batch_ids(
     data_context: Optional["DataContext"] = None,  # noqa: F821
+    batch_list: Optional[List[Batch]] = None,
     batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict, str]] = None,
     domain: Optional[Domain] = None,
     variables: Optional[ParameterContainer] = None,
     parameters: Optional[Dict[str, ParameterContainer]] = None,
 ) -> Optional[List[str]]:
-    if batch_request is None:
-        return None
+    batch: Batch
+    if batch_list is None or all([batch is None for batch in batch_list]):
+        if batch_request is None:
+            return None
```

Comment:
Same here and throughout.

Reply:
@cdkini Cannot use the same line, because if the condition if batch_request is None: is False, we must not return, but continue on to the next statement (to obtain batch_request by building it differently). Thanks.